docs/general/ddl/data-types.rst

```diff
@@ -2799,8 +2799,11 @@ This means ``TEXT[]`` is equivalent to ``ARRAY(text)``.
 
 .. NOTE::
 
-    Nested arrays are not supported. Something like ``ARRAY(ARRAY(TEXT))``
-    won't work.
+    Nested arrays are not supported as a column type. Something like
+    ``ARRAY(ARRAY(TEXT))`` won't work. However, it's possible to store a
+    multidimensional array as a part of an object.
```

Comment:
Can also add some example like
```
CREATE TABLE tbl (obj OBJECT(ignored));
INSERT INTO tbl (obj) VALUES ({two_dim_array = [[1,2,3],[4,5,6]]});   
```
but not sure it's needed here

Reply:
I have the impression that the added text is not all that useful. A user reading this might assume that you can use `object as (xs int[][])` or something like that, but that doesn't work. Instead it's only that you can have `array(object as (xs int[]))`.

I'm not sure if there's much value in pointing this out here.
I'd leave the docs as they are and we can at some point in the future look into enabling nested arrays. I think similar to group by on arrays the restriction is mostly artificial by now.